### ISSUE TITLE
Implement Custom Elements v1

### ISSUE THREAD

**Issue by g4jc** [MoonchildProductions/UXP#1344]
We have most of Custom Elements v0 in the tree, this was an evolving spec that has since moved on to v1. In order to be compatible with websites using Custom Elements v1, we need to update existing code.

https://developers.google.com/web/fundamentals/web-components/customelements

Blocks #252

**Comment by wolfbeast** [comment]
Looks like we got lucky with Mozilla having something half-finished in our tree for once.
Can you indicate in which order all this is supposed to land? Does #1343 block this?

**Comment by g4jc** [comment]
> Can you indicate in which order all this is supposed to land? Does #1343 block this?

Yes, while it may be possible to implement CE without <dialog>, landing #1343 first makes it much easier. Both cause changes in the HTML Tree Binding, so this will land later.

**Comment by mattatobin** [comment]
Blocks #1361

**Comment by g4jc** [comment]
We definitely still need to land [
customElements.define must upgrade custom element in shadow-including tree order
](https://bugzilla.mozilla.org/show_bug.cgi?id=1326028) for web  compat. Unfortunately without CSS selectors, testing will be problematic.
